rust/lance/src/index/vector/flat.rs

```diff
@@ -72,7 +75,8 @@ pub async fn flat_search(stream: impl RecordBatchStream, query: &Query) -> Resul
         .await?;
 
     if batches.is_empty() {
-        return Ok(RecordBatch::new_empty(input_schema));
+        let schema_with_distance = input_schema.try_with_column(distance_field())?;
+        return Ok(RecordBatch::new_empty(schema_with_distance.into()));
```

Comment:
Does you know if we include the row id on empty tables if the user asks for the row id?

Reply:
for this one, I think row id will either come from the `Take` node. We won't have to handle appending row id here. I'll check if `Take` handles this correctly.